src/DataStructures.jl

```diff
@@ -94,4 +94,6 @@ module DataStructures
     export status
     export deref_key, deref_value, deref, advance, regress
 
+    @deprecate iter(s::Stack) collect(s)
+    @deprecate iter(q::Queue) collect(q)
```

Comment:
we could explicitly call depwarn, but the intended behavior of `iter` was to provide an iterable collection not an array copy of the contents of the stack / queue so it wouldn't be unreasonable to break the latter. I wonder if it would make sense to PR to `Base` another `deprecate` macro that took a third argument with a custom deprecation message.

Either we'd need to use `collect` but provide clear deprecation warnings stating that you should just switch from `iter(s)` to `s` if you're using this for iteration but switch from `iter(s)` to `collect(s)` if you're just looking for a copy of the contents of `s` or we would need to switch to `s` and accept that this may potentially break existing use cases. FWIW, this method wasn't out in the wild for horribly long and I have doubts that people were using it for getting copies of the contents of the stack, but happy to hear additional opinions on this.